.github/workflows/ngen_integration.yaml

```diff
@@ -0,0 +1,139 @@
+        run: |
+          cmake -B cmake_build -S . -DNGEN=ON
+          make -C cmake_build
+
+      - name: Save SoilFreezeThaw to a Temp Directory
+        run: |
+          # Move files to a temporary directory
+          mkdir ${{runner.temp}}/sft
+          mv ./* ${{runner.temp}}/sft
+
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - name: checkout actions in another repo
+        uses: actions/checkout@v4
+        with:
+          repository: noaa-owp/ngen
+
+      - name: Build Surfacebmi
+        id: submod_build_1
+        uses: ./.github/actions/ngen-submod-build
+        with:
+          mod-dir: " extern/noah-owp-modular/"
+          targets: "surfacebmi"
+
+      - name: Build ISO C Fortran BMI
+        id: submod_build_2
+        uses: ./.github/actions/ngen-submod-build
+        with:
+          mod-dir: "extern/iso_c_fortran_bmi/"
+
+      - name: Build CFE
+        id: submod_build_3
+        uses: ./.github/actions/ngen-submod-build
+        with:
+          mod-dir: "extern/cfe/"
+          targets: "cfebmi"
+
+      - name: Build PET
+        id: submod_build_5
+        uses: ./.github/actions/ngen-submod-build
+        with:
+          mod-dir: "extern/evapotranspiration/evapotranspiration"
+          targets: "petbmi"
+
+      - name: Build SLoTH
+        id: submod_build_6
+        uses: ./.github/actions/ngen-submod-build
+        with:
+          mod-dir: "extern/sloth/"
+          targets: "slothmodel"
+
+      # Build Ngen and save some files
+      - name: Build Ngen
+        id: ngen_id1
+        uses: ./.github/actions/ngen-build
+        with:
+          targets: "ngen"
+          build-cores: ${{ env.LINUX_NUM_PROC_CORES }}
+          #is this required for this test?
+          bmi_c: 'ON'
+          bmi_fortran: 'ON'
+        timeout-minutes: 15
+
+      - name: Move SFT Files Including cmake_build to Appropriate Directory
+        run: |
+          rm -rf extern/SoilFreezeThaw/SoilFreezeThaw/*
+          mv ${{runner.temp}}/sft/* extern/SoilFreezeThaw/SoilFreezeThaw
+
+      - name: Move SMP Files Including cmake_build to Appropriate Directory
+        run: |
+          rm -rf extern/SoilMoistureProfiles/SoilMoistureProfiles/*
+          mv ${{runner.temp}}/smp/* extern/SoilMoistureProfiles/SoilMoistureProfiles
+
+      - name: Run Ngen Test
+        run: |
+          # Run ngen with topmodel with CFE
+          mv ${{ steps.ngen_id1.outputs.build-dir }} ./ngen-build/
+          inputfile='extern/SoilMoistureProfiles/SoilMoistureProfiles/config/realization_config_smp_ngenCI.json'
+          ./ngen-build/ngen ./data/catchment_data.geojson "cat-27" ./data/nexus_data.geojson "nex-26" $inputfile
+
+      # The following remove a post ngen build error likely related to the temporary files left over during the
+      # build process
+      - name: Re-checkout Ngen
+        uses: actions/checkout@v3
+        with:
+          repository: noaa-owp/ngen
```

Comment:
Do you have any additional context/logs for the error that this solves?

Reply:
I have removed the re-checkout part of the code and now it passes the automated test.